### PR TITLE
restrict archived suggestions to parent goals only and fix subgoal render problem

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
Fixed:

- only show suggestion of archived child goals of a parent
- fix re-render problem
- show different toast msg when suggested goal is added.

### CODE REVIEW

**Review by tijlleenders — APPROVED** [review]
Great, PR does what it says it does!

File: src/api/GoalsAPI/index.ts [hunk 1]
```diff
@@ -353,14 +353,27 @@ export const addHintGoaltoMyGoals = async (goal: GoalItem) => {
   ]);
 };
 
-export const fetchArchivedGoalByTitle = async (value: string) => {
-  if (value.trim() === "") {
+export const fetchArchivedGoalByTitle = async (goalId: string, parentGoalId: string) => {
+  if (goalId.trim() === "") {
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:357]: try this => if( !goalId.trim() )

File: src/api/GoalsAPI/index.ts [hunk 2]
```diff
@@ -353,14 +353,27 @@ export const addHintGoaltoMyGoals = async (goal: GoalItem) => {
   ]);
 };
 
-export const fetchArchivedGoalByTitle = async (value: string) => {
-  if (value.trim() === "") {
+export const fetchArchivedGoalByTitle = async (goalId: string, parentGoalId: string) => {
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:356]: if we say byTitle then arg should be named as title. can we change that goalId to goalTitle or title?

File: src/api/GoalsAPI/index.ts [hunk 3]
```diff
@@ -353,14 +353,27 @@ export const addHintGoaltoMyGoals = async (goal: GoalItem) => {
...
     return [];
   }
-  const lowerCaseValue = value.toLowerCase();
-  return db.goalsCollection
-    .where("archived")
-    .equals("true")
-    .and((goal) => goal.title.toLowerCase().startsWith(lowerCaseValue))
-    .toArray();
+
+  const lowerCaseValue = goalId.toLowerCase();
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:361]: maybe we can rename it something like searchTitle ?

File: src/api/GoalsAPI/index.ts [hunk 4]
```diff
@@ -353,14 +353,27 @@ export const addHintGoaltoMyGoals = async (goal: GoalItem) => {
...
+
+  const getAllChildrenGoals = async (id: string): Promise<GoalItem[]> => {
+    const directChildren = await getChildrenGoals(id);
+
+    const allChildren = await Promise.all(
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:366]: maybe name it to -> allDescendantGoals

File: src/api/GoalsAPI/index.ts [hunk 5]
```diff
@@ -353,14 +353,27 @@ export const addHintGoaltoMyGoals = async (goal: GoalItem) => {
...
+      directChildren.map(async (child) => {
+        const grandchildren = await getAllChildrenGoals(child.id);
+        return [child, ...grandchildren];
+      }),
+    );
+
+    return allChildren.flat();
+  };
+
+  const allChildren = await getAllChildrenGoals(parentGoalId);
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:376]: allDescendantGoals

File: src/contexts/parentGoal-context.tsx [hunk 6]
```diff
@@ -67,6 +69,12 @@ export const ParentGoalProvider = ({ children }: { children: ReactNode }) => {
     init();
   }, [parentId]);
 
+  useEffect(() => {
+    if (suggestedGoal === null) {
```
Comment (Tushar-4781) [inline src/contexts/parentGoal-context.tsx:73]: if (!suggestedGoal) {

File: src/hooks/useGoalActions.tsx [hunk 7]
```diff
@@ -90,11 +92,20 @@ const useGoalActions = () => {
...
       setLastAction("goalUpdated");
-      setShowToast({
-        open: true,
-        message: "Goal updated!",
-        extra: "",
-      });
+      if (suggestedGoal) {
+        setShowToast({
+          open: true,
+          message: "Goal (re)created!",
```
Comment (Tushar-4781) [inline src/hooks/useGoalActions.tsx:98]: u can just use ternary operator for the message field

File: src/hooks/useGoalStore.tsx [hunk 8]
```diff
@@ -1,21 +1,26 @@
...
   const { partnerId } = useParams();
   const navigate = useNavigate();
   const location = useLocation();
   const showConfirmation = useRecoilValue(displayConfirmation);
 
-  const openEditMode = (goal: GoalItem) => {
+  const openEditMode = (goal: GoalItem, customState?: ILocationState) => {
     const prefix = `${partnerId ? `/partners/${partnerId}/` : "/"}goals`;
+
+    const newState = {
```
Comment (Tushar-4781) [inline src/hooks/useGoalStore.tsx:16]: no need of variable u can place directly in there

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**